src/signals-mach.c

```diff
@@ -501,39 +424,7 @@ void *mach_profile_listener(void *arg)
             unw_context_t *uc;
             jl_thread_suspend_and_get_state(i, &uc);
             if (running) {
-#ifdef LIBOSXUNWIND
-                /*
-                 *  Unfortunately compact unwind info is incorrectly generated for quite a number of
-                 *  libraries by quite a large number of compilers. We can fall back to DWARF unwind info
-                 *  in some cases, but in quite a number of cases (especially libraries not compiled in debug
-                 *  mode, only the compact unwind info may be available). Even more unfortunately, there is no
-                 *  way to detect such bogus compact unwind info (other than noticing the resulting segfault).
-                 *  What we do here is ugly, but necessary until the compact unwind info situation improves.
-                 *  We try to use the compact unwind info and if that results in a segfault, we retry with DWARF info.
-                 *  Note that in a small number of cases this may result in bogus stack traces, but at least the topmost
-                 *  entry will always be correct, and the number of cases in which this is an issue is rather small.
-                 *  Other than that, this implementation is not incorrect as the other thread is paused while we are profiling
-                 *  and during stack unwinding we only ever read memory, but never write it.
-                 */
-
-                forceDwarf = 0;
```

Comment:
I assume this removal means they got their act together on unwind quality?

Reply:
I need to confirm this yet but I haven't noticed any segfaults and `unw_init_local_dwarf` doesn't exist in the LLVM libunwind.

Reply:
I've now observed some issues with using LLVM libunwind with this logic removed. I also noticed that `unw_init_local_dwarf` is one of your creations :) https://github.com/JuliaLang/libosxunwind/commit/bb95a5cf07ac245c3dadfa33254f34f80a9d2cd9

Reply:
There are still some issues with unwind quality but the force DWARF fix you made doesn't fix the problem as it did with libosxunwind: https://github.com/JuliaLang/julia/pull/39127#issuecomment-761872940

I've reintroduced this logic.